src/eddymotion/registration/utils.py

```diff
@@ -0,0 +1,114 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""
+Utilities to aid in performing and evaluating image registration.
+
+This module provides functions to compute displacements of image coordinates
+under a transformation, useful for assessing the accuracy of image registration
+processes.
+
+"""
+
+from __future__ import annotations
+
+from itertools import product
+
+import nibabel as nb
+import nitransforms as nt
+import numpy as np
+
+
+def displacements_within_mask(
+    maskimg: nb.spatialimages.SpatialImage,
```

Comment:
You have used ":class:`~nibabel.spatialimages.SpatialImage`" in
https://github.com/nipreps/eddymotion/pull/225/files#diff-097d3fe0cd1687fc7df869ef3ace9c3dd40ae4e9d292a97ee02d45686bbe6ce9R428

Also, if we prefer to adopt `nb.spatialimages.SpatialImage` rather than `nibabel.nifti1.Nifti1Image`, we may need to revisit other docstrings.

Reply:
Ah, yes, my bad, not awake yet. But in the docstrings sometimes you have used ":class:`~`" vs ":obj:`~`".

Reply:
:obj: is for when you don't want to break your brain and think the exact type of reference. The ~ symbol just tells sphinx to hide the full module tree and show only the name.